plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/Pom.java

```diff
@@ -87,8 +88,31 @@ static Pom parse(IFile pomFile) throws SAXException, IOException, CoreException
       DocumentBuilder builder = builderFactory.newDocumentBuilder();
       Document document = builder.parse(pomFile.getContents());
       Pom pom = new Pom(document, pomFile);
+      
+      XPath xpath = xpathFactory.newXPath();
+      xpath.setNamespaceContext(new Maven4NamespaceContext());
+
+      NodeList bomNodes = (NodeList) xpath.evaluate(
+          "//m:dependencyManagement/m:dependencies/m:dependency",
```

Comment:
Actually, I just got this question. Is everything in this section is actually a BOM?

Reply:
Every dependency element

Reply:
BOMs are [`<type>pom</type><scope>import</scope>`](https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Importing_Dependencies).
> *import* (only available in Maven 2.0.9 or later)
> This scope is only supported on a dependency of type pom in the <dependencyManagement> section. It indicates the dependency to be replaced with the effective list of dependencies in the specified POM's <dependencyManagement> section. Since they are replaced, dependencies with a scope of import do not actually participate in limiting the transitivity of a dependency.

The non-import dependencies matter, but they aren't BOMs.  For example, [we define a dependencyManagement section](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/blob/master/pom.xml#L159) in our pom.xml.

Reply:
Okay, sounds like we need to check the type and the scope before passing it to `Bom.loadBom()`.